File: Java-Python-SQL-Fundamentals/Java Fundamentals/Inheritance-Practice3/src/main/java/org/example/Account.java
package org.example;

//Parent
public class Account {

    private String username;
    private String email;
    private boolean isLoggedIn;


    
    public Account(String username, String email) {

        this.username = username;
        this.email = email;
        isLoggedIn = false;
    }


    public void login() {
        if (isLoggedIn) {
            System.out.println("Already logged in");
            return;
        }
        System.out.println("Logging in....\n" + "Welcome " + username);
        isLoggedIn = true;

    }

    public void logout() {
        if (!isLoggedIn) {
            System.out.println("Already logged out");
            return;
        }

        System.out.println("Logging out...");
        isLoggedIn = false;

    }

    protected boolean isUserLoggedIn() {
        return isLoggedIn;
    }

    public String getAccountInfo() {

        return "Account details:\n" +
                "Username: " + username +
                "\nEmail: " + email +
                "\nLogin Status: " + isLoggedIn+
                "\n";
    }


}

File: Java-Python-SQL-Fundamentals/Python-Fundamentals/Python_OOP_Inheritance2/Account.py
class MainAccount:

    def __init__(self, username, email):

        self.username = username
        self.email = email
        self.logged_in = False

    def login(self):

        if self.logged_in:
            print('Already logged in')
            return

        print('Logging in...')
        print(f'Welcome {self.username}')
        self.logged_in = True

    def logout(self):

        if not self.logged_in:
            print('Not logged in')
            return

        print('logging out...')
        self.logged_in = False

    def _is_logged_in(self):
        return self.logged_in

    def get_account_info(self) -> str:

        return ('Account Details: \n'
                f'Username: {self.username} \n'
                f'Email: {self.email} \n'
                f'Login Status: {self.logged_in}'

                )
